Board.h

```diff
@@ -0,0 +1,28 @@
+#ifndef BOARD_H
+#define BOARD_H
+
+#include "Snake.h"
+
+class Board
+{
+public:
+    Board() = default;
+
+    int& operator()(int x, int y) { return board[x][y]; }
```

Comment:
it's not obvious what this operator is supposed to do
i'd rather add some explicit method such as 'getCellValue' or something like that

also you return reference that gives access to your board from external world
it would better to just return the value from one function and modify it from another (getter and setter)